examples/shopping-cart-cqrs/akka-shopping-cart-streamlet/src/main/scala/shoppingcart/streamlets/ShoppingCartStreamlet.scala

```diff
@@ -0,0 +1,100 @@
+import scala.concurrent.duration._
+import cloudflow.akkastream.{AkkaStreamlet, Clustering}
+import sample.cqrs.ShoppingCart.{CheckedOut, ItemAdded, ItemQuantityAdjusted, ItemRemoved}
+import sample.cqrs.ShoppingCart
+import shoppingcart.cqrs.{EventProcessorSettings, Main}
+import shoppingcart.data.ShoppingCartEvent
+
+class ShoppingCartStreamlet extends AkkaStreamlet with Clustering {
+  val out   = AvroOutlet[ShoppingCartEvent]("out")
+  val shape = StreamletShape.withOutlets(out)
+
+  override def createLogic = new RunnableGraphStreamletLogic() {
+    val settings = EventProcessorSettings(system.toTyped)
+    ShoppingCart.init(system.toTyped, settings)
+    val sharding = ClusterSharding(system.toTyped)
+
+    Main.startNode(system.toTyped)
+
+    val shardedDaemonProcessSettings =
+      ShardedDaemonProcessSettings(system.toTyped)
+
+    ShardedDaemonProcess(system.toTyped).init(
+      name = "ShoppingCartProjection",
+      settings.parallelism,
+      n => ProjectionBehavior(createProjectionFor(system.toTyped, settings, n)),
+      shardedDaemonProcessSettings,
+      Some(ProjectionBehavior.Stop))
+
+    implicit val timeout:Timeout = 5.seconds
+
+    def runnableGraph = Source.repeat(UUID.randomUUID().toString)
+      .throttle(1, 2.second)
+      .mapAsync(1)(id => {
+        val entityRef =
+          sharding.entityRefFor(ShoppingCart.EntityKey, id)
+        entityRef.ask[ShoppingCart.Confirmation](ref => ShoppingCart.AddItem("socks",1, ref))
+          .map(x=> {
+            entityRef.ask[ShoppingCart.Confirmation](ref => ShoppingCart.AdjustItemQuantity("socks", 10, ref))
+          })
+      })
+      .to(Sink.ignore)
+
+    def createProjectionFor(
+                             system: ActorSystem[_],
+                             settings: EventProcessorSettings,
+                             index: Int
+                           ): AtLeastOnceFlowProjection[Offset, EventEnvelope[ShoppingCart.Event]] = {
+      val tag = s"${settings.tagPrefix}-$index"
+      val sourceProvider = EventSourcedProvider.eventsByTag[ShoppingCart.Event](
+        system = system,
+        readJournalPluginId = CassandraReadJournal.Identifier,
+        tag = tag)
+
+      val flow = FlowWithContext[EventEnvelope[ShoppingCart.Event], ProjectionContext]
+        .map(x => {
+          println("Event source message: "+x)
+          x
+        })
+        .map( x =>
+          x.event match {
+            case e: ItemAdded => ShoppingCartEvent(e.cartId, e.itemId, e.quantity, 0, "ItemAdded")
+            case e: ItemRemoved => ShoppingCartEvent(e.cartId, e.itemId, 0, 0, "ItemRemoved")
+            case e: ItemQuantityAdjusted => ShoppingCartEvent(e.cartId, e.itemId, e.newQuantity, 0, "ItemQuantityAdjusted")
+            case e: CheckedOut => ShoppingCartEvent(e.cartId, "", 0, System.currentTimeMillis(), "CheckedOut")
+            case _ => ShoppingCartEvent("", "", 0, 0, "Error")
+          })
+        .mapAsync(1) { x =>
+          sinkRef(out).write(x)
```

Comment:
@RayRoestenburg and @ennru this `sinkRef(out).write(x)` from the cloudflow api is what I am using to be able write a message to the cloudflow outlet before akka projections commits the offset.

This function uses the `KafkaProducer(...).send( )` from the Kafka client to write to kafka.  

For this example would it be better to use an AkkaProjection actor, Source.queue, and a MergeHub?  I was looking at the KafkaProducer documentation and I am pretty sure this operation can handle many parallel streams calling it at once but I wanted to hear your thoughts.